src/main/java/beam/analysis/plots/RealizedModeAnalysis.java

```diff
@@ -625,6 +630,56 @@ public void writeToReferenceCSV(){
         }
     }
 
+    private void writeToReplaningChainCSV(IterationEndsEvent event) {
+        String fileName = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(event.getIteration(), "replanningEventChain.csv");
+        String REPLANNING_SEPARATOR = "-"+ReplanningEvent.EVENT_TYPE+"-";
+        try (BufferedWriter out = new BufferedWriter(new FileWriter(new File(fileName)))) {
+            String heading = "modeChoiceReplanningEventChain,count";
+            out.write(heading);
+            out.newLine();
+            Set<String> persons = personReplanningChain.keySet();
+            Map<String, Integer> modeCount = new HashedMap();
+            for(String person: persons){
+                List<String> modes = personReplanningChain.get(person);
+                if(modes.size() > 1){
+                    StringBuffer lastMode = new StringBuffer();
+                    boolean isLastReplaning = false;
+                    for(String mode: modes){
+                        if(ReplanningEvent.EVENT_TYPE.equals(mode)){
```

Comment:
Is this supposed to be an if else, or just an if to set the variable? Additionally, is this the best way to tell if it is the last? I guess I don't know this data structuring well enough, but it seems open to a lot of bugs. Would more assertions be beneficial? How about a test or tests to make sure this works as expected?

Reply:
Hi @JustinPihony I have simplifid the code. Also last means all modes that come before "Replanning" event. Added couple of test cases to cover this scenerio.